File: Python/1_Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d = {}
        for i in range(len(nums)):
            if nums[i] not in d:
                d[nums[i]] = [i]
            else:
                d[nums[i]].append(i)
        for i in range(len(nums)):
            if target-nums[i] in d:
                if target - nums[i] == nums[i]:
                    if len(d[target - nums[i]]) > 1:
                        return [d[nums[i]][0], d[nums[i]][1]]
                else:
                    return [i, d[target-nums[i]][0]]
        

File: C++/1_Two_Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, vector<int>> a;
        vector<int> ans;
        for(int i = 0; i < nums.size(); i++){
            a[nums[i]].push_back(i);
        }
        for(int i = 0; i < nums.size(); i++){
            if(a.count(target - nums[i]) == 1){
                if(target - nums[i] == nums[i]){
                    if(a[target-nums[i]].size() > 1){
                        ans = {a[nums[i]][0], a[nums[i]][1]};
                        break;
                    }
                } else {
                    ans = {i, a[target-nums[i]][0]};
                    break;
                }
            }
        }
        return ans;
    }
};